bin/bank/pycbc_brute_bank

```diff
@@ -278,29 +312,54 @@ def draw(rtype):
     params = {}
 
     if rtype == 'uniform':
-        for name, pmin, pmax in zip(args.params, args.min, args.max):
-            params[name] = numpy.random.uniform(pmin, pmax, size=size)
+        if args.input_config is None:
+            for name, pmin, pmax in zip(args.params, args.min, args.max):
+                params[name] = numpy.random.uniform(pmin, pmax, size=size)
+        else:
+            # `draw_samples_from_config` has its own fixed seed, so must overwrite it.
+            random_seed = numpy.random.randint(low=0, high=2**32-1)
+            samples = draw_samples_from_config(args.input_config, size, random_seed)
+            for name in samples.fieldnames:
+                params[name] = samples[name]
+
     elif rtype == 'kde':
         trail = 300
         if trail > len(bank):
             trail = len(bank)
         p = bank.keys()
         p = [k for k in p if k not in fdict]
         p.remove('approximant')
+        if args.input_config is not None and 'q' in static_args:
+            # For the fixed mass ratio, it will encounter `singular matrix` issue
+            # when calculate the covariance matrix in KDE method, so ignore it.
+            p.remove('mass1')
+            p.remove('mass2')
         bdata = numpy.array([bank.key(k)[-trail:] for k in p])
 
         kde = gaussian_kde(bdata)
         points = kde.resample(size=size)
         for k, v in zip(p, points):
             params[k] = v
 
+        if args.input_config is not None and 'q' in static_args:
+            params['mass1'] = pycbc.conversions.mass1_from_mtotal_q(params['total_mass'], static_args['q'])
+            params['mass2'] = pycbc.conversions.mass2_from_mtotal_q(params['total_mass'], static_args['q'])
+
     params['approximant'] = numpy.array([args.approximant]*size)
 
-    # Filter out stuff
+    # Filter out stuff (kde method may also generate samples outside boundaries).
```

Comment:
If you have the config file, this isn't the right way to way to apply the filter. There is a much easier method that already handles both boundaries *and* constraints. Just use the contains operator of the JointDistribution. . 


Look at https://github.com/gwastro/pycbc/blob/master/pycbc/distributions/joint.py#L220